third_party/2and3/sqlalchemy/sql/functions.pyi

```diff
@@ -1,47 +1,144 @@
+# Stubs for sqlalchemy.sql.functions (Python 3.6)
+#
+# NOTE: This dynamically typed stub was automatically generated by stubgen.
 
-from .base import Executable, ColumnCollection
-from .elements import ClauseList, Cast, Extract, _literal_as_binds, \
-    literal_column, _type_from_args, ColumnElement, _clone,\
-    Over, BindParameter, FunctionFilter
-from .selectable import FromClause, Select, Alias
-from .visitors import VisitableType
+from typing import Any, Optional
+from .base import Executable as Executable, ColumnCollection as ColumnCollection
+from .elements import ClauseList as ClauseList, Cast as Cast, Extract as Extract, literal_column as literal_column, ColumnElement as ColumnElement, Over as Over, BindParameter as BindParameter, FunctionFilter as FunctionFilter, Grouping as Grouping, WithinGroup as WithinGroup
+from .selectable import FromClause as FromClause, Select as Select, Alias as Alias
+from . import util as sqlutil
+from .visitors import VisitableType as VisitableType
 
-class FunctionElement(Executable, ColumnElement, FromClause): ...
+def register_function(identifier, fn, package: str = ...): ...
+
+class FunctionElement(Executable, ColumnElement, FromClause):
+    packagenames = ...  # type: Any
+    clause_expr = ...  # type: Any
+    def __init__(self, *clauses, **kwargs) -> None: ...
+    @property
+    def columns(self): ...
+    def clauses(self): ...
+    def over(self, partition_by: Optional[Any] = ..., order_by: Optional[Any] = ..., rows: Optional[Any] = ..., range_: Optional[Any] = ...): ...
+    def within_group(self, *order_by): ...
+    def filter(self, *criterion): ...
+    def get_children(self, **kwargs): ...
+    def within_group_type(self, within_group): ...
+    def alias(self, name: Optional[Any] = ..., flat: bool = ...): ...
+    def select(self): ...
+    def scalar(self): ...
+    def execute(self): ...
+    def self_group(self, against: Optional[Any] = ...): ...
 
 class _FunctionGenerator(object):
-    def __init__(self, **opts): ...
+    opts = ...  # type: Any
+    def __init__(self, **opts) -> None: ...
     def __getattr__(self, name): ...
-    def __call__(self, *c, **kwargs) -> Function: ...
+    def __call__(self, *c, **kwargs): ...
+
+func = ...  # type: Any
+modifier = ...  # type: Any
+
+class Function(FunctionElement):
+    __visit_name__ = ...  # type: str
+    packagenames = ...  # type: Any
+    name = ...  # type: Any
+    type = ...  # type: Any
+    def __init__(self, name, *clauses, **kw) -> None: ...
+
+class _GenericMeta(VisitableType):
+    def __init__(cls, clsname, bases, clsdict) -> None: ...
 
-func = ...  # type: _FunctionGenerator
-modifier = ...  # type: _FunctionGenerator
+class GenericFunction(object):
```

Comment:
Super should be `Function` not `object`. The aut-generated chain is probably lost [around here](https://github.com/zzzeek/sqlalchemy/blob/master/lib/sqlalchemy/sql/functions.py#L457):

```python
class GenericFunction(util.with_metaclass(_GenericMeta, Function)):
    ...
```

Otherwise we get type incompatibility errors for things like:

```python
x: sa.ColumnElement
y: sa.ColumnElement
x = sa.sum(y)
```

Where we try to assign `sum` type to `sa.ColumnElement`.